chia/cmds/plotnft_funcs.py

```diff
@@ -262,86 +274,98 @@ async def submit_tx_with_confirmation(
+        if enforce_https and not pool_url.startswith("https://"):
+            print(f"Pool URLs must be HTTPS on mainnet {pool_url}. Aborting.")
+            return
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(
+                    f"{pool_url}/pool_info", ssl=ssl_context_for_root(get_mozilla_ca_crt())
+                ) as response:
+                    if response.ok:
+                        json_dict = json.loads(await response.text())
+                    else:
+                        print(f"Response not OK: {response.status}")
+                        return
+        except Exception as e:
+            print(f"Error connecting to pool {pool_url}: {e}")
+            return
 
-    msg = f"Will start self-farming with Plot NFT on wallet id {wallet_id} fingerprint {fingerprint}."
-    func = functools.partial(wallet_client.pw_self_pool, wallet_id, fee_mojos)
-    await submit_tx_with_confirmation(msg, prompt, func, wallet_client, fingerprint, wallet_id)
+        if json_dict["relative_lock_height"] > 1000:
+            print("Relative lock height too high for this pool, cannot join")
+            return
+        if json_dict["protocol_version"] != POOL_PROTOCOL_VERSION:
+            print(f"Incorrect version: {json_dict['protocol_version']}, should be {POOL_PROTOCOL_VERSION}")
+            return
 
+        pprint(json_dict)
+        msg = f"\nWill join pool: {pool_url} with Plot NFT {fingerprint}."
+        func = functools.partial(
+            wallet_client.pw_join_pool,
+            wallet_id,
+            hexstr_to_bytes(json_dict["target_puzzle_hash"]),
+            pool_url,
+            json_dict["relative_lock_height"],
+            fee_mojos,
+        )
 
-async def inspect_cmd(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    wallet_id = args.get("id", None)
-    pool_wallet_info, unconfirmed_transactions = await wallet_client.pw_status(wallet_id)
-    print(
-        {
-            "pool_wallet_info": pool_wallet_info,
-            "unconfirmed_transactions": [
-                {"sent_to": tx.sent_to, "transaction_id": tx.name.hex()} for tx in unconfirmed_transactions
-            ],
-        }
-    )
+        await submit_tx_with_confirmation(msg, prompt, func, wallet_client, fingerprint, wallet_id)
 
 
-async def claim_cmd(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    wallet_id = args.get("id", None)
-    fee = Decimal(args.get("fee", 0))
-    fee_mojos = uint64(int(fee * units["chia"]))
-    msg = f"\nWill claim rewards for wallet ID: {wallet_id}."
-    func = functools.partial(
-        wallet_client.pw_absorb_rewards,
-        wallet_id,
-        fee_mojos,
-    )
-    await submit_tx_with_confirmation(msg, False, func, wallet_client, fingerprint, wallet_id)
+async def self_pool(
+    wallet_rpc_port: Optional[int], fingerprint: int, fee: Decimal, wallet_id: int, prompt: bool
```

Comment:
I think it would be good to force callers to name the parameters here too, with the `*,`